model/sys/vmem_utils.sail

```diff
@@ -106,7 +106,66 @@ function misaligned_order(n) = {
   }
 }
 
-// External API
+// Pointer Masking
+
+// Return the 2-bit PMM value based on the current effective privilege mode.
+function get_pmm(eff_privilege : Privilege) -> PointerMaskingMode = match eff_privilege {
+  Machine    => pmm_mode(mseccfg[PMM]),
+  Supervisor => pmm_mode(menvcfg[PMM]),
+  User       => if currentlyEnabled(Ext_S) then pmm_mode(senvcfg[PMM]) else pmm_mode(menvcfg[PMM]),
+  VirtualUser       => internal_error(__FILE__, __LINE__, "Hypervisor extension not supported"),
+  VirtualSupervisor => internal_error(__FILE__, __LINE__, "Hypervisor extension not supported"),
+}
+
+// Check whether PMM is applicable.
+// NOTE: mstatus[MXR] == 1'b1 disables pointer masking in S and U mode, but not in M-mode.
+function is_pmm_applicable(typ : MemoryAccessType(ext_access_type), eff_privilege : Privilege) -> bool =
+  typ != InstructionFetch() & (eff_privilege == Machine | mstatus[MXR] == 0b0) & xlen == 64
```

Comment:
Is this actually correct? I don't quite recall the conclusion of the discussion about MXR on the mailing list but that sounds surprising to me that MXR would disable it for S and U mode? I thought MXR existed to allow M-mode to load from execute-only pages, so I don't see why it should affect non-M-mode? 

Actually reading the name of the parameters again, does this only apply when MPRV is set? If so we might want to add a comment.